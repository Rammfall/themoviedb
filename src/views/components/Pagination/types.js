import PropTypes from 'prop-types'

export const paginationPropTypes = {
  current: PropTypes.number,
  total: PropTypes.number,
  onChange: PropTypes.func.isRequired
}

export const paginationDefaultProps = {
  current: 1,
  total: 1
}
