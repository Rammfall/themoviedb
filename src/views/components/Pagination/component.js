import React from 'react'
import PropTypes from 'prop-types'
import { Col, Row, Pagination as AntPagination } from 'antd'
import { useHistory } from 'react-router-dom'

import { paginationQuantityPerPage } from 'Constants/index'

const Pagination = ({ current, total, onChange }) => {
  const history = useHistory()

  return (
    <Row
      type='flex'
      justify='center'
    >
      <Col>
        <AntPagination
          defaultCurrent={1}
          current={current}
          total={total}
          defaultPageSize={paginationQuantityPerPage}
          className='pagination'
          onChange={(page) => {
            history.push(`?page=${page}`)
            onChange(page)
          }}
        />
      </Col>
    </Row>
  )
}

Pagination.propTypes = {
  current: PropTypes.number,
  total: PropTypes.number,
  onChange: PropTypes.func.isRequired
}

Pagination.defaultProps = {
  current: 1,
  total: 1
}

export default Pagination
