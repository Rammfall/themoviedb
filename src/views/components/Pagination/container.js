import React from 'react'
import PropTypes from 'prop-types'
import { useHistory } from 'react-router-dom'

import changePage from 'Utils/router/changePage'
import queryParams from 'Utils/router/queryParams'

import PaginationComponent from './component'

const Pagination = ({ total }) => {
  const history = useHistory()
  const onPageChange = changePage(history)
  const current = Number(queryParams(history.location.search).get('page')) || 1

  return (
    <PaginationComponent
      current={current}
      total={total}
      onChange={onPageChange}
    />
  )
}

Pagination.propTypes = {
  total: PropTypes.number
}

Pagination.defaultProps = {
  total: 0
}
export default Pagination
