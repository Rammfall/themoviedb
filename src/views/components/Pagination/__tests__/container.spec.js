import React from 'react'
import { shallow } from 'enzyme'

import Pagination from '../container'

jest.mock('react-router-dom', () => ({
  useHistory: jest.fn()
}))

describe('Pagination', () => {
  it('matches snapshot', () => {
    const container = shallow(<Pagination onChange={jest.fn()} />)

    expect(container).toMatchSnapshot()
  })
})
