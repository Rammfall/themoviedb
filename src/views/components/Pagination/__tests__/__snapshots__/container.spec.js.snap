// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination with page param matches snapshot 1`] = `
<PaginationComponent
  current={1}
  onChange={[MockFunction]}
  total={0}
/>
`;

exports[`Pagination without page param matches snapshot 1`] = `
<PaginationComponent
  current={1}
  onChange={[MockFunction]}
  total={0}
/>
`;
